### ISSUE TITLE
ICE: 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:931:9

### ISSUE THREAD

**Issue by rtyler** [rust-lang/rust#68372]
This may have a duplicate, since there are similar panics already reported, but this is on a different line number.

The compilation log and source code are listed below

```rust
pub mod client;
pub mod msg;

use std::collections::HashMap;
use std::sync::Arc;
use tokio::sync::broadcast::{channel, Receiver, Sender};

/**
 * A channel is named and typed with the type of messages it should be carrying
 */
struct Channel {
    name: String,
    send: Sender<Message>,
    recv: Receiver<Message>,
}

struct Message {}

impl Channel {
    fn send(&self, msg: Message) {}
    fn recv(&self, msg: Message) {}
}

struct Bus {
    /**
     * Channels are named and can implement a number of different types. This should
     * allow the Bus to handle different channels with different message payloads
     * while still taking advantage of compile-time checks
     */
    channels: HashMap<String, Channel>,
}

#[cfg(test)]
mod tests {
    use super::*;
}
```

[compile.log](https://github.com/rust-lang/rust/files/4083271/compile.log)

**Comment by Centril** [comment]
```
Script started on 2020-01-19 10:59:24-08:00 [TERM="tmux-256color" TTY="/dev/pts/11" COLUMNS="127" LINES="51"]
k..urce/git/otto\[0m[27m[24m[J[01;32m➜  [36motto[00m [01;34mgit:([31mwarp-speed[34m) [33m✗[00m [K[?1h=[?2004hRRUST_BACKTRACE=1 cargo build --verbose[?1l>[?2004l

kcargo\[0m[0m[1m[32m       Fresh[0m autocfg v0.1.7
[0m[0m[1m[32m       Fresh[0m cfg-if v0.1.10
[0m[0m[1m[32m       Fresh[0m unicode-xid v0.2.0
[0m[0m[1m[32m       Fresh[0m version_check v0.1.5
[0m[0m[1m[32m       Fresh[0m rand_core v0.4.2
[0m[0m[1m[32m       Fresh[0m itoa v0.4.4
[0m[0m[1m[32m       Fresh[0m fnv v1.0.6
[0m[0m[1m[32m       Fresh[0m lazy_static v1.4.0
[0m[0m[1m[32m       Fresh[0m siphasher v0.2.3
[0m[0m[1m[32m       Fresh[0m bytes v0.5.3
[0m[0m[1m[32m       Fresh[0m futures-core v0.3.1
[0m[0m[1m[32m       Fresh[0m ppv-lite86 v0.2.6
[0m[0m[1m[32m       Fresh[0m semver-parser v0.7.0
[0m[0m[1m[32m       Fresh[0m version_check v0.9.1
[0m[0m[1m[32m       Fresh[0m slab v0.4.2
[0m[0m[1m[32m       Fresh[0m futures-sink v0.3.1
[0m[0m[1m[32m       Fresh[0m matches v0.1.8
[0m[0m[1m[32m       Fresh[0m arc-swap v0.4.4
[0m[0m[1m[32m       Fresh[0m quick-error v1.2.2
[0m[0m[1m[32m       Fresh[0m smallvec v1.0.0
[0m[0m[1m[32m       Fresh[0m byte-tools v0.3.1
[0m[0m[1m[32m       Fresh[0m same-file v1.0.5
[0m[0m[1m[32m       Fresh[0m regex-syntax v0.6.12
[0m[0m[1m[32m       Fresh[0m ucd-trie v0.1.2
[0m[0m[1m[32m       Fresh[0m futures-task v0.3.1
[0m[0m[1m[32m       Fresh[0m pin-project-lite v0.1.1
[0m[0m[1m[32m       Fresh[0m pin-utils v0.1.0-alpha.4
[0m[0m[1m[32m       Fresh[0m percent-encoding v2.1.0
[0m[0m[1m[32m       Fresh[0m opaque-debug v0.2.3
[0m[0m[1m[32m       Fresh[0m maplit v1.0.2
[0m[0m[1m[32m       Fresh[0m serde v0.8.23
[0m[0m[1m[32m       Fresh[0m nodrop v0.1.14
[0m[0m[1m[32m       Fresh[0m fake-simd v0.1.2
[0m[0m[1m[32m       Fresh[0m unicode-width v0.1.7
[0m[0m[1m[32m       Fresh[0m dtoa v0.4.4
[0m[0m[1m[32m       Fresh[0m try-lock v0.2.2
[0m[0m[1m[32m       Fresh[0m safemem v0.3.3
[0m[0m[1m[32m       Fresh[0m remove_dir_all v0.5.2
[0m[0m[1m[32m       Fresh[0m linked-hash-map v0.5.2
[0m[0m[1m[32m       Fresh[0m mime v0.3.14
[0m[0m[1m[32m       Fresh[0m static_assertions v0.3.4
[0m[0m[1m[32m       Fresh[0m base64 v0.11.0
[0m[0m[1m[32m       Fresh[0m futures-io v0.3.1
[0m[0m[1m[32m       Fresh[0m tower-service v0.3.0
[0m[0m[1m[32m       Fresh[0m termcolor v1.0.5
[0m[0m[1m[32m       Fresh[0m utf-8 v0.7.5
[0m[0m[1m[32m       Fresh[0m void v1.0.2
[0m[0m[1m[32m       Fresh[0m vec_map v0.8.1
[0m[0m[1m[32m       Fresh[0m strsim v0.8.0
[0m[0m[1m[32m       Fresh[0m urlencoding v1.0.0
[0m[0m[1m[32m       Fresh[0m utf8parse v0.1.1
[0m[0m[1m[32m       Fresh[0m rust-ini v0.13.0
[0m[0m[1m[32m       Fresh[0m ansi_term v0.11.0
[0m[0m[1m[32m       Fresh[0m unicode-segmentation v1.6.0
[0m[0m[1m[32m       Fresh[0m scoped-tls v1.0.0
[0m[0m[1m[32m       Fresh[0m yaml-rust v0.3.5
[0m[0m[1m[32m       Fresh[0m rand_core v0.3.1
[0m[0m[1m[32m       Fresh[0m rand_jitter v0.1.4
[0m[0m[1m[32m       Fresh[0m thread_local v0.3.6
[0m[0m[1m[32m       Fresh[0m http v0.2.0
[0m[0m[1m[32m       Fresh[0m c2-chacha v0.2.3
[0m[0m[1m[32m       Fresh[0m semver v0.9.0
[0m[0m[1m[32m       Fresh[0m futures-channel v0.3.1
[0m[0m[1m[32m       Fresh[0m unicode-bidi v0.3.4
[0m[0m[1m[32m       Fresh[0m humantime v1.3.0
[0m[0m[1m[32m       Fresh[0m block-padding v0.1.5
[0m[0m[1m[32m       Fresh[0m unicode-normalization v0.1.11
[0m[0m[1m[32m       Fresh[0m walkdir v2.2.9
[0m[0m[1m[32m       Fresh[0m pest v2.1.2
[0m[0m[1m[32m       Fresh[0m futures-util v0.3.1
[0m[0m[1m[32m       Fresh[0m serde_test v0.8.23
[0m[0m[1m[32m       Fresh[0m textwrap v0.11.0
[0m[0m[1m[32m       Fresh[0m yaml-rust v0.4.3
[0m[0m[1m[32m       Fresh[0m libc v0.2.66
[0m[0m[1m[32m       Fresh[0m proc-macro2 v1.0.6
[0m[0m[1m[32m       Fresh[0m log v0.4.8
[0m[0m[1m[32m       Fresh[0m memchr v2.2.1
[0m[0m[1m[32m       Fresh[0m rand_xorshift v0.1.1
[0m[0m[1m[32m       Fresh[0m rand_isaac v0.1.1
[0m[0m[1m[32m       Fresh[0m rand_hc v0.1.0
[0m[0m[1m[32m       Fresh[0m byteorder v1.3.2
[0m[0m[1m[32m       Fresh[0m typenum v1.11.2
[0m[0m[1m[32m       Fresh[0m headers-core v0.2.0
[0m[0m[1m[32m       Fresh[0m http-body v0.3.1
[0m[0m[1m[32m       Fresh[0m ryu v1.0.2
[0m[0m[1m[32m       Fresh[0m rustc_version v0.2.3
[0m[0m[1m[32m       Fresh[0m bitflags v1.2.1
[0m[0m[1m[32m       Fresh[0m httparse v1.3.4
[0m[0m[1m[32m       Fresh[0m idna v0.2.0
[0m[0m[1m[32m       Fresh[0m rust-embed-utils v5.0.0
[0m[0m[1m[32m       Fresh[0m pest_meta v2.1.2
[0m[0m[1m[32m       Fresh[0m arrayvec v0.4.12
[0m[0m[1m[32m       Fresh[0m futures v0.3.1
[0m[0m[1m[32m       Fresh[0m linked-hash-map v0.3.0
[0m[0m[1m[32m       Fresh[0m quote v1.0.2
[0m[0m[1m[32m       Fresh[0m iovec v0.1.4
[0m[0m[1m[32m       Fresh[0m rand_os v0.1.3
[0m[0m[1m[32m       Fresh[0m getrandom v0.1.13
[0m[0m[1m[32m       Fresh[0m net2 v0.2.33
[0m[0m[1m[32m       Fresh[0m time v0.1.42
[0m[0m[1m[32m       Fresh[0m num_cpus v1.11.1
[0m[0m[1m[32m       Fresh[0m signal-hook-registry v1.2.0
[0m[0m[1m[32m       Fresh[0m dirs v1.0.5
[0m[0m[1m[32m       Fresh[0m atty v0.2.13
[0m[0m[1m[32m       Fresh[0m dirs-sys v0.3.4
[0m[0m[1m[32m       Fresh[0m rand_chacha v0.1.1
[0m[0m[1m[32m       Fresh[0m rand_pcg v0.1.2
[0m[0m[1m[32m       Fresh[0m num-traits v0.2.10
[0m[0m[1m[32m       Fresh[0m indexmap v1.3.0
[0m[0m[1m[32m       Fresh[0m aho-corasick v0.7.6
[0m[0m[1m[32m       Fresh[0m log v0.3.9
[0m[0m[1m[32m       Fresh[0m buf_redux v0.8.4
[0m[0m[1m[32m       Fresh[0m twoway v0.1.8
[0m[0m[1m[32m       Fresh[0m want v0.3.0
[0m[0m[1m[32m       Fresh[0m unicase v1.4.2
[0m[0m[1m[32m       Fresh[0m generic-array v0.12.3
[0m[0m[1m[32m       Fresh[0m unicase v2.6.0
[0m[0m[1m[32m       Fresh[0m nix v0.14.1
[0m[0m[1m[32m       Fresh[0m url v2.1.0
[0m[0m[1m[32m       Fresh[0m syn v1.0.11
[0m[0m[1m[32m       Fresh[0m bytes v0.4.12
[0m[0m[1m[32m       Fresh[0m rand_core v0.5.1
[0m[0m[1m[32m       Fresh[0m mio v0.6.21
[0m[0m[1m[32m       Fresh[0m shellexpand v1.1.0
[0m[0m[1m[32m       Fresh[0m dirs v2.0.2
[0m[0m[1m[32m       Fresh[0m clap v2.33.0
[0m[0m[1m[32m       Fresh[0m rand v0.6.5
[0m[0m[1m[32m       Fresh[0m num-traits v0.1.43
[0m[0m[1m[32m       Fresh[0m num-integer v0.1.41
[0m[0m[1m[32m       Fresh[0m regex v1.3.1
[0m[0m[1m[32m       Fresh[0m mime v0.2.6
[0m[0m[1m[32m       Fresh[0m phf_shared v0.7.24
[0m[0m[1m[32m       Fresh[0m block-buffer v0.7.3
[0m[0m[1m[32m       Fresh[0m digest v0.8.1
[0m[0m[1m[32m       Fresh[0m serde_derive v1.0.104
[0m[0m[1m[32m       Fresh[0m rand_chacha v0.2.1
[0m[0m[1m[32m       Fresh[0m tokio-macros v0.2.3
[0m[0m[1m[32m       Fresh[0m pin-project-internal v0.4.6
[0m[0m[1m[32m       Fresh[0m http v0.1.21
[0m[0m[1m[32m       Fresh[0m input_buffer v0.2.0
[0m[0m[1m[32m       Fresh[0m pest_generator v2.1.1
[0m[0m[1m[32m       Fresh[0m mio-uds v0.6.7
[0m[0m[1m[32m       Fresh[0m rust-embed-impl v5.2.0
[0m[0m[1m[32m       Fresh[0m rustyline v5.0.6
[0m[0m[1m[32m       Fresh[0m serde-hjson v0.9.1
[0m[0m[1m[32m       Fresh[0m env_logger v0.6.2
[0m[0m[1m[32m       Fresh[0m phf_generator v0.7.24
[0m[0m[1m[32m       Fresh[0m sha-1 v0.8.1
[0m[0m[1m[32m       Fresh[0m phf v0.7.24
[0m[0m[1m[32m       Fresh[0m serde v1.0.104
[0m[0m[1m[32m       Fresh[0m lexical-core v0.4.6
[0m[0m[1m[32m       Fresh[0m rand v0.7.2
[0m[0m[1m[32m       Fresh[0m pin-project v0.4.6
[0m[0m[1m[32m       Fresh[0m pest_derive v2.1.0
[0m[0m[1m[32m       Fresh[0m tokio v0.2.6
[0m[0m[1m[32m       Fresh[0m rust-embed v5.2.0
[0m[0m[1m[32m       Fresh[0m phf_codegen v0.7.24
[0m[0m[1m[32m       Fresh[0m serde_json v1.0.44
[0m[0m[1m[32m       Fresh[0m toml v0.5.5
[0m[0m[1m[32m       Fresh[0m chrono v0.4.10
[0m[0m[1m[32m       Fresh[0m hashbrown v0.5.0
[0m[0m[1m[32m       Fresh[0m serde_urlencoded v0.6.1
[0m[0m[1m[32m       Fresh[0m headers v0.3.0
[0m[0m[1m[32m       Fresh[0m serde_yaml v0.8.11
[0m[0m[1m[32m       Fresh[0m tempfile v3.1.0
[0m[0m[1m[32m       Fresh[0m tungstenite v0.9.2
[0m[0m[1m[32m       Fresh[0m mime_guess v2.0.1
[0m[0m[1m[32m       Fresh[0m nom v5.0.1
[0m[0m[1m[32m       Fresh[0m uuid v0.8.1
[0m[0m[1m[32m       Fresh[0m tokio-util v0.2.0
[0m[0m[1m[32m       Fresh[0m pretty_env_logger v0.3.1
[0m[0m[1m[32m       Fresh[0m handlebars v2.0.2
[0m[0m[1m[32m       Fresh[0m h2 v0.2.1
[0m[0m[1m[32m       Fresh[0m config v0.10.1
[0m[0m[1m[32m       Fresh[0m mime_guess v1.8.7
[0m[0m[1m[32m       Fresh[0m hyper v0.13.1
[0m[0m[1m[32m       Fresh[0m multipart v0.16.1
[0m[0m[1m[32m       Fresh[0m warp v0.2.0
[0m[0m[1m[32m   Compiling[0m otto-eventbus v0.1.0 (/home/tyler/source/git/otto/eventbus)
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K[0m[0m[1m[32m     Running[0m `rustc --edition=2018 --crate-name otto_eventbus eventbus/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 -C metadata=0d23839fb1da08c1 -C extra-filename=-0d23839fb1da08c1 --out-dir /home/tyler/source/git/otto/target/debug/deps -C incremental=/home/tyler/source/git/otto/target/debug/incremental -L dependency=/home/tyler/source/git/otto/target/debug/deps --extern chrono=/home/tyler/source/git/otto/target/debug/deps/libchrono-d52802bffc8a82b9.rmeta --extern clap=/home/tyler/source/git/otto/target/debug/deps/libclap-4771d1530caa7736.rmeta --extern config=/home/tyler/source/git/otto/target/debug/deps/libconfig-3ec895d80071174d.rmeta --extern futures=/home/tyler/source/git/otto/target/debug/deps/libfutures-f8d371ca64b9b3ea.rmeta --extern handlebars=/home/tyler/source/git/otto/target/debug/deps/libhandlebars-9c39757c14b838fa.rmeta --extern log=/home/tyler/source/git/otto/target/debug/deps/liblog-79b125b8f2be9618.rmeta --extern pretty_env_logger=/home/tyler/source/git/otto/target/debug/deps/libpretty_env_logger-00a4f0333544d957.rmeta --extern rust_embed=/home/tyler/source/git/otto/target/debug/deps/librust_embed-2606570b3ffd432e.rmeta --extern rustyline=/home/tyler/source/git/otto/target/debug/deps/librustyline-6ad46db1b00fa316.rmeta --extern serde=/home/tyler/source/git/otto/target/debug/deps/libserde-a8e99c52ea927756.rmeta --extern serde_json=/home/tyler/source/git/otto/target/debug/deps/libserde_json-2aecfbb27590c90b.rmeta --extern tokio=/home/tyler/source/git/otto/target/debug/deps/libtokio-a5aa0c1f2c9a1372.rmeta --extern warp=/home/tyler/source/git/otto/target/debug/deps/libwarp-8b1a4c7665c21343.rmeta -C link-arg=-fuse-ld=lld`
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K[0m[1m[38;5;9merror: internal compiler error[0m[0m[1m: src/librustc/dep_graph/graph.rs:680: DepNode Hir(otto_eventbus[fe62]::Channel[0]::send[0]) should have been pre-allocated but wasn't.[0m

[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[Kthread 'rustc' panicked at 'Box<Any>', src/librustc_errors/lib.rs:931:9
stack backtrace:
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.40/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:77
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:61
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1028
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1412
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:65
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:50
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:188
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:205
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  10: rustc_driver::report_ice
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  11: <alloc::boxed::Box<F> as core::ops::function::Fn<A>>::call
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/liballoc/boxed.rs:956
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  12: proc_macro::bridge::client::<impl proc_macro::bridge::Bridge>::enter::{{closure}}::{{closure}}
             at /rustc/73528e339aae0f17a15ffa49a8ac608f50c6cf14/src/libproc_macro/bridge/client.rs:305
  13: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:468
  14: std::panicking::begin_panic
  15: rustc_errors::HandlerInner::bug
  16: rustc_errors::Handler::bug
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  17: rustc::util::bug::opt_span_bug_fmt::{{closure}}
  18: rustc::ty::context::tls::with_opt::{{closure}}
  19: rustc::ty::context::tls::with_context_opt
  20: rustc::ty::context::tls::with_opt
  21: rustc::util::bug::opt_span_bug_fmt
  22: rustc::util::bug::bug_fmt
  23: rustc::dep_graph::graph::DepGraph::try_mark_previous_green
  24: rustc::dep_graph::graph::DepGraph::try_mark_green
  25: rustc::dep_graph::graph::DepGraph::try_mark_green_and_read
  26: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  27: <rustc_typeck::outlives::implicit_infer::InferVisitor as rustc::hir::itemlikevisit::ItemLikeVisitor>::visit_item
  28: rustc::hir::Crate::visit_all_item_likes
  29: rustc_typeck::outlives::inferred_outlives_crate
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  30: rustc::ty::query::__query_compute::inferred_outlives_crate
  31: rustc::dep_graph::graph::DepGraph::with_task_impl
  32: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  33: rustc_typeck::outlives::inferred_outlives_of
  34: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::inferred_outlives_of>::compute
  35: rustc::dep_graph::graph::DepGraph::with_task_impl
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  36: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::force_query
  37: rustc::ty::query::plumbing::force_from_dep_node
  38: rustc::dep_graph::graph::DepGraph::try_mark_previous_green
  39: rustc::dep_graph::graph::DepGraph::try_mark_previous_green
  40: rustc::dep_graph::graph::DepGraph::try_mark_green
  41: rustc::dep_graph::graph::DepGraph::try_mark_green_and_read
  42: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  43: <rustc_typeck::collect::CollectItemTypesVisitor as rustc::hir::intravisit::Visitor>::visit_item
  44: rustc::hir::map::Map::visit_item_likes_in_module
  45: rustc_typeck::collect::collect_mod_item_types
  46: rustc::ty::query::__query_compute::collect_mod_item_types
  47: rustc::ty::query::<impl rustc::ty::query::config::QueryAccessors for rustc::ty::query::queries::collect_mod_item_types>::compute
  48: rustc::dep_graph::graph::DepGraph::with_task_impl
  49: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  50: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::ensure_query
  51: rustc_typeck::check_crate::{{closure}}::{{closure}}
  52: rustc::util::common::time
  53: rustc_typeck::check_crate
  54: rustc_interface::passes::analysis
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K  55: rustc::ty::query::__query_compute::analysis
  56: rustc::dep_graph::graph::DepGraph::with_task_impl
  57: rustc::ty::query::plumbing::<impl rustc::ty::context::TyCtxt>::get_query
  58: rustc_interface::passes::BoxedGlobalCtxt::access::{{closure}}
  59: rustc_interface::passes::create_global_ctxt::{{closure}}
  60: rustc_interface::passes::BoxedGlobalCtxt::enter
  61: rustc_interface::interface::run_compiler_in_existing_thread_pool
  62: std::thread::local::LocalKey<T>::with
  63: scoped_tls::ScopedKey<T>::set
  64: syntax::with_globals
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.

[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[Knote: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: rustc 1.40.0 (73528e339 2019-12-16) running on x86_64-unknown-linux-gnu

note: compiler flags: -C debuginfo=2 -C incremental -C link-arg=-fuse-ld=lld --crate-type lib

note: some of the compiler flags provided by cargo are hidden

query stack during panic:
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K#0 [type_of] processing `Channel::send`
#1 [inferred_outlives_crate] computing the inferred outlives predicates for items in this crate
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K#2 [inferred_outlives_of] processing `Channel`
#3 [predicates_of] processing `Channel`
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K#4 [collect_mod_item_types] collecting item types in top-level module
#5 [analysis] running analysis passes on this crate
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[Kend of query stack
[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K[0m[1m[38;5;9merror[0m[0m[1m: aborting due to previous error[0m

[0m[0m[1m[36m    Building[0m [====================================================>  ] 230/235: otto-eventbus                                
[K[0m[0m[1m[31merror[0m[1m:[0m could not compile `otto-eventbus`.

Caused by:
  process didn't exit successfully: `rustc --edition=2018 --crate-name otto_eventbus eventbus/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 -C metadata=0d23839fb1da08c1 -C extra-filename=-0d23839fb1da08c1 --out-dir /home/tyler/source/git/otto/target/debug/deps -C incremental=/home/tyler/source/git/otto/target/debug/incremental -L dependency=/home/tyler/source/git/otto/target/debug/deps --extern chrono=/home/tyler/source/git/otto/target/debug/deps/libchrono-d52802bffc8a82b9.rmeta --extern clap=/home/tyler/source/git/otto/target/debug/deps/libclap-4771d1530caa7736.rmeta --extern config=/home/tyler/source/git/otto/target/debug/deps/libconfig-3ec895d80071174d.rmeta --extern futures=/home/tyler/source/git/otto/target/debug/deps/libfutures-f8d371ca64b9b3ea.rmeta --extern handlebars=/home/tyler/source/git/otto/target/debug/deps/libhandlebars-9c39757c14b838fa.rmeta --extern log=/home/tyler/source/git/otto/target/debug/deps/liblog-79b125b8f2be9618.rmeta --extern pretty_env_logger=/home/tyler/source/git/otto/target/debug/deps/libpretty_env_logger-00a4f0333544d957.rmeta --extern rust_embed=/home/tyler/source/git/otto/target/debug/deps/librust_embed-2606570b3ffd432e.rmeta --extern rustyline=/home/tyler/source/git/otto/target/debug/deps/librustyline-6ad46db1b00fa316.rmeta --extern serde=/home/tyler/source/git/otto/target/debug/deps/libserde-a8e99c52ea927756.rmeta --extern serde_json=/home/tyler/source/git/otto/target/debug/deps/libserde_json-2aecfbb27590c90b.rmeta --extern tokio=/home/tyler/source/git/otto/target/debug/deps/libtokio-a5aa0c1f2c9a1372.rmeta --extern warp=/home/tyler/source/git/otto/target/debug/deps/libwarp-8b1a4c7665c21343.rmeta -C link-arg=-fuse-ld=lld` (exit code: 101)
k..urce/git/otto\[0m[27m[24m[J[01;31m➜  [36motto[00m [01;34mgit:([31mwarp-speed[34m) [33m✗[00m [K[?1h=[?2004h[?2004l


Script done on 2020-01-19 10:59:33-08:00 [COMMAND_EXIT_CODE="101"]
```

**Comment by Centril** [comment]
Does this reproduce with the beta or nightly compiler?

Also, if possible, a self-contained reproducer without `tokio` (i.e. an example that fits in the playground) would be helpful towards fixing the ICE.

**Comment by hellow554** [comment]
@rtyler can you provide the sources for the complete project? Do you have a git repository which we could check out and reproduce the bug?

**Comment by rtyler** [comment]
@hellow554 This project is open source, and I [checkpointed for this issue](https://git.brokenco.de/rtyler/otto/commit/d837eb8af1a0d69ff0d8079fc6a2f00236abf2a8).

A `git clone https://git.brokenco.de/rtyler/otto` will get the repo for you 

Unfortunately I was not able to take the time for a larger diversion into inspection of this issue.

**Comment by hellow554** [comment]
I need some further explanation here @rtyler 
I tried stable, beta and nightly, none of them crashed. I used `cargo build`.
What is your compiler (`rustc --version`) and what do you need to crash it (e.g. do I need to do something beforehand?). Also, does `cargo clean` solve your problem?

**Comment by rtyler** [comment]
@hellow554 the version is 1.4.0, and full clean did appear to correct the issue (unfortunately). I'm not sure what more can be done since the issue isn't terribly reproducible.

I would have thought my `Cargo.lock` would have made this more reliably reproducible, but alas :pouting_cat:

**Comment by hellow554** [comment]
Seems to be an incremental issue then? @Centril what do you think?

**Comment by pnkfelix** [comment]
This I believe will be resolved by PR #68289.

triage: P-high, removing nomination label, assigning to self.

**Comment by pnkfelix** [comment]
Since PR #68289 has landed, and there isn't enough information here to reproduce the bug as originally described (and thus I cannot test to see whether it is actually fixed, or create a regression test), I'm going to close this issue under the assumption that PR #68289 did fix the problem here.
